tests/checkpoint/test_checkpoint.py

```diff
@@ -2337,6 +2337,7 @@ def test_newstyle_checkpoint_config_substitution_nested(
         CheckpointConfig(
             name="my_nested_checkpoint",
             config_version=1,
+            template_name="my_nested_checkpoint_template_3",
```

Comment:
Is there a new test to be added for the bug you referenced?

Reply:
@NathanFarmer This test has effectively been added by repairing some existing tests.  In addition, the functionality will be fully manifested in a near-future PR that focuses on `checkpoint.run` payload instrumentation, which is how the bug was detected in the first place (the `batch_data` values were empty in the `runtime_parameter` dictionary).  Thanks.